chore: upgrade to 2024 edition

JP-Ellis opened the pull request:
Signed-off-by: JP-Ellis <josh@jpellis.me>

JP-Ellis committed:
chore: upgrade to 2024 edition

Signed-off-by: JP-Ellis <josh@jpellis.me>

JP-Ellis committed:
chore(deps): upgrade dependencies

Signed-off-by: JP-Ellis <josh@jpellis.me>

JP-Ellis commented:
btw, this is a first PR. I'll follow up with other PRs to address #153 and then implement #165

olix0r commented on rust-toolchain:
```diff
@@ -1 +1 @@
-1.64.0
+stable
```
i think we probably want to update this to a recent static version -- this appears to break fetching rustfmt, at least.

olix0r commented:
Actually, we probably want to update this to be a rust-toolchain.toml…

Oliver Gould < ***@***.*** >

On Thu, Sep 18, 2025 at 8:06 AM, Oliver Gould < ***@***.*** > wrote:

> 
> 
> 
> ***@***.**** commented on this pull request.
> 
> 
> 
> In rust-toolchain (
> https://github.com/olix0r/cargo-action-fmt/pull/166#discussion_r2359783699
> ) :
> 
> > @@ -1 +1 @@
> -1.64.0
> +stable
> 
> 
> i think we probably want to update this to a recent static version -- this
> appears to break fetching rustfmt, at least.
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub (
> https://github.com/olix0r/cargo-action-fmt/pull/166#pullrequestreview-3240351839
> ) , or unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AAB2YYWO5BHSFAEUO3LGB433TLDA3AVCNFSM6AAAAACGQFUWGSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTENBQGM2TCOBTHE
> ).
> You are receiving this because you are subscribed to this thread. Message
> ID: <olix0r/cargo-action-fmt/pull/166/review/3240351839 @ github. com>
> 
> 
>

JP-Ellis commented:
Do you have a strong preference for pinning the toolchain?

My preference would be to remove the Rust toolchain file entirely, and instead update CI to test against both `stable` and `beta` (in case of upcoming breaking changes).

If you want to target a MSRV, we can also add that into the mix.

JP-Ellis committed:
chore: remove rust-toolchain

The new CI runs tests against stable and beta versions of Rust.

Signed-off-by: JP-Ellis <josh@jpellis.me>

JP-Ellis committed:
chore: update deny.toml to latest version

Update the `deny.toml` file to be compatible with the latest version of
`cargo-deny`. I have also taken the liberty of removing a few fields
which matched the defaults.

Signed-off-by: JP-Ellis <josh@jpellis.me>

JP-Ellis committed:
chore(ci): rework CI

Combine the deps and lint pipelines into one, with a final job to check
that everything passed fine.

Also adjust the CI tasks to only run on `main` or PRs targetting `main`.

Signed-off-by: JP-Ellis <josh@jpellis.me>
